packages/model-registry/upstream/frontend/src/app/pages/modelRegistry/screens/RegisteredModels/RegisteredModelTable.tsx

```diff
@@ -4,40 +4,65 @@ import { ModelVersion, RegisteredModel } from '~/app/types';
 import { getLatestVersionForRegisteredModel } from '~/app/pages/modelRegistry/screens/utils';
 import { rmColumns } from './RegisteredModelsTableColumns';
 import RegisteredModelTableRow from './RegisteredModelTableRow';
+import { useModelDeploymentDetection } from '~/odh/utils/deploymentUtils';
 
 type RegisteredModelTableProps = {
   clearFilters: () => void;
   registeredModels: RegisteredModel[];
   modelVersions: ModelVersion[];
   refresh: () => void;
+  getModelDeploymentInfo?: (rmId: string) => { hasDeploys: boolean; loaded: boolean };
 } & Partial<Pick<React.ComponentProps<typeof Table>, 'toolbarContent'>>;
 
-const RegisteredModelTable: React.FC<RegisteredModelTableProps> = ({
+const RegisteredModelTableContent: React.FC<RegisteredModelTableProps> = ({
   clearFilters,
   registeredModels,
   modelVersions,
   toolbarContent,
   refresh,
-}) => (
-  <Table
-    data-testid="registered-model-table"
-    data={registeredModels}
-    columns={rmColumns}
-    toolbarContent={toolbarContent}
-    defaultSortColumn={2}
-    onClearFilters={clearFilters}
-    enablePagination
-    emptyTableView={<DashboardEmptyTableView onClearFilters={clearFilters} />}
-    rowRenderer={(rm: RegisteredModel) => (
-      <RegisteredModelTableRow
-        key={rm.name}
-        hasDeploys={false}
-        registeredModel={rm}
-        latestModelVersion={getLatestVersionForRegisteredModel(modelVersions, rm.id)}
-        refresh={refresh}
-      />
-    )}
-  />
-);
+  getModelDeploymentInfo,
+}) => {
+  const { hasRegisteredModelDeployment, loaded } = useModelDeploymentDetection();
+  
+  const hasDeploysForModel = React.useCallback((rmId: string) => {
+    const { hasDeployment } = hasRegisteredModelDeployment(rmId, modelVersions);
+    return hasDeployment;
+  }, [hasRegisteredModelDeployment, modelVersions]);
+
+  return (
+    <Table
+      data-testid="registered-model-table"
+      data={registeredModels}
+      columns={rmColumns}
+      toolbarContent={toolbarContent}
+      defaultSortColumn={2}
+      onClearFilters={clearFilters}
+      enablePagination
+      emptyTableView={<DashboardEmptyTableView onClearFilters={clearFilters} />}
+      rowRenderer={(rm: RegisteredModel) => {
+        const deploymentInfo = getModelDeploymentInfo?.(rm.id) ?? { 
+          hasDeploys: hasDeploysForModel(rm.id), 
+          loaded 
+        };
+        
+        return (
+          <RegisteredModelTableRow
+            key={rm.name}
+            hasDeploys={deploymentInfo.hasDeploys}
+            loaded={deploymentInfo.loaded}
+            registeredModel={rm}
+            latestModelVersion={getLatestVersionForRegisteredModel(modelVersions, rm.id)}
+            refresh={refresh}
+          />
+        );
+      }}
+    />
+  );
+};
+
+const RegisteredModelTable: React.FC<RegisteredModelTableProps> = (props) => {
+  // Use deployment context from parent (RegisteredModelListView)
+  return <RegisteredModelTableContent {...props} />;
+};
```

Comment:
This is a pretty considerable diff, and the RegisteredModelTable wrapper isn't doing anything. I wonder if we can change this so we don't have it touching every line of the rendered `<Table>`. Can we move the hook calls and `hasRegisteredModelDeployment` stuff into your wrapper, and maybe just have the wrapper component create and pass in an alternate rowRenderer function that will be used instead if present?